### PR TITLE
schnorrsig: Add test vectors for variable-length messages 

### PR DESCRIPTION
**Pull request by real-or-random**
Related: https://github.com/sipa/bips/pull/221

### CODE REVIEW

**Comment by real-or-random** [comment]
Updated after one vector was removed from my draft at https://github.com/sipa/bips/pull/221.

I think this is ready for reviewm/merge, independently of whether https://github.com/sipa/bips/pull/221 has been merged or merged in the official BIPs repo.

File: include/secp256k1_schnorrsig.h [hunk 1]
```diff
@@ -141,8 +141,12 @@ SECP256K1_API int secp256k1_schnorrsig_sign(
  *  variable length messages and accepts a pointer to an extraparams object that
  *  allows customizing signing by passing additional arguments.
  *
- *  Creates the same signatures as schnorrsig_sign if msglen is 32 and the
- *  extraparams.ndata is the same as aux_rand32.
+ *  Equivalent to secp256k1_schnorrsig_sign32(..., auxrand32) if msglen is 32
```
Comment (jonasnick) [inline include/secp256k1_schnorrsig.h:144]: auxrand32 -> aux_rand32

File: src/modules/schnorrsig/tests_impl.h [hunk 2]
```diff
@@ -241,28 +241,31 @@ void test_schnorrsig_sha256_tagged(void) {
...
+void test_schnorrsig_bip_vectors_check_signing(const unsigned char *sk, const unsigned char *pk_serialized, const unsigned char *aux_rand, const unsigned char *msg, size_t msglen, const unsigned char *expected_sig) {
     unsigned char sig[64];
     secp256k1_keypair keypair;
     secp256k1_xonly_pubkey pk, pk_expected;
 
+    secp256k1_schnorrsig_extraparams extraparams = SECP256K1_SCHNORRSIG_EXTRAPARAMS_INIT;
+    extraparams.ndata = (unsigned char*)aux_rand;
+
     CHECK(secp256k1_keypair_create(ctx, &keypair, sk));
-    CHECK(secp256k1_schnorrsig_sign32(ctx, sig, msg32, &keypair, aux_rand));
```
Comment (jonasnick) [inline src/modules/schnorrsig/tests_impl.h:250]: Wouldn't it be better if we continued running `secp256k1_schnorrsig_sign32` if the msglen is 32?

Comment (real-or-random) [reply src/modules/schnorrsig/tests_impl.h]: > Or run both in that case.

done

**Review by jonasnick — COMMENTED** [review]
ACK cd54ac7c1cca509404b62e626a6291f434af88e8

**Pull request merged**